src/sys/stat.rs

```diff
@@ -402,14 +403,51 @@ pub fn lutimes<P: ?Sized + NixPath>(
     Errno::result(res).map(drop)
 }
 
-/// Change the access and modification times of the file specified by a file descriptor.
+/// A helper function to convert `atime: Option<&TimeSpec>, mtime: Option<&TimeSpec>`
+/// to `[libc::timespec; 2], used in the implementation of [`utimnsat(3)`]
+/// and [`futimens(3)`]
+fn time_convert(
+    atime: Option<&TimeSpec>,
+    mtime: Option<&TimeSpec>,
+) -> [libc::timespec; 2] {
+    let mut times = [
+        libc::timespec {
+            tv_sec: 0,
+            tv_nsec: libc::UTIME_OMIT,
+        },
+        libc::timespec {
+            tv_sec: 0,
+            tv_nsec: libc::UTIME_OMIT,
+        },
```

Comment:
```suggestion
        libc::timespec {
            tv_sec: 0,
            tv_nsec: libc::UTIME_OMIT,
            ..
        },
        libc::timespec {
            tv_sec: 0,
            tv_nsec: libc::UTIME_OMIT,
            ..
        },
```
See #1886 for the reason why.